### ISSUE TITLE
Duplicate tests stanzas runs without warning

### ISSUE THREAD

**Issue by FintanH** [commercialhaskell/stack#4159]
### General summary/comments
This may be a `hpack` issue, but since I discovered it through stack I thought I'd post it here. 

During some development I discovered that a `package.yaml` file had multiple `tests` stanzas. This meant that the second test suite would run but the other would not. Obviously this is bad :) It would be nice if this was caught and an error reported or possibly have the two stanzas merge.

### Steps to reproduce

You can reproduce by cloning https://github.com/FintanH/duplicate-stanza and running `stack test`.

### Expected
```
error: Duplicate "tests" stanzas declared in duplicate-stanza-0.1.0.0
```

or

```
warning: Duplicate "tests" stanzas declared in duplicate-stanza-0.1.0.0, attempting to merge them.

duplicate-stanza-0.1.0.0: test (suite: first)
             
Progress: 1/2first: This never runs
CallStack (from HasCallStack):
  error, called at test/first/Spec.hs:2:8 in main:Main
             
duplicate-stanza-0.1.0.0: Test suite first failed
duplicate-stanza-0.1.0.0: test (suite: second)
             
Whoo tests passed

duplicate-stanza-0.1.0.0: Test suite second passed
Completed 2 action(s).
Test suite failure for package duplicate-stanza-0.1.0.0
    first:  exited with: ExitFailure 1
Logs printed to console
```

### Actual

```
$ stack test
duplicate-stanza-0.1.0.0: configure (test)
Configuring duplicate-stanza-0.1.0.0...
duplicate-stanza-0.1.0.0: build (test)
Preprocessing test suite 'second' for duplicate-stanza-0.1.0.0..
Building test suite 'second' for duplicate-stanza-0.1.0.0..
duplicate-stanza-0.1.0.0: test (suite: second)
             
Progress: 1/2Whoo tests passed
             
duplicate-stanza-0.1.0.0: Test suite second passed
Completed 2 action(s).
```

### Stack version

```
$ stack --version
Version 1.7.1, Git revision 681c800873816c022739ca7ed14755e85a579565 (5807 commits) x86_64 hpack-0.28.2
```

### Method of installation

* Official binary, downloaded from stackage.org or fpcomplete's package repository

**Comment by mbj** [comment]
I think the issue is in the yaml parser that "silently" collapses keys (last occurence wins). Curious if the yaml package provides a way to inspect the AST before such a collapse happens, or if its possible to instruct to fail on duplicates.

**Comment by dbaynard** [comment]
@sol's response in @quasicomputational's comment suggests this won't be fixed in `hpack`… unless the behaviour of `yaml` changes.

From a cursory glance, I've been unable to find a relevant issue on the `yaml` tracker, so it may be a good idea to raise this upstream — @quasicomputational would you be willing to do that?

Meanwhile, I'll bring this to @snoyberg's attention — I can see how it might be a desirable change. If he's willing to support changes upstream, the changes will cascade (via the linked `hpack` issue) to stack. If he's not, then it's a won't fix. In either case, I'm closing for now.

**Comment by sol** [comment]
I'm looking into his.

**Comment by dbaynard** [comment]
Do raise the issue with yaml, @quasicomputational, as the maintainer is open to such changes.

**Comment by FintanH** [comment]
Thanks for the feedback 👌 💯

**Comment by sol** [comment]
This will be addressed by https://github.com/sol/hpack/pull/319.

**Comment by dbaynard** [comment]
I'm reopening to make sure the hpack change gets in to the new release of stack, if possible.

**Comment by k-bx** [comment]
Sorry for making a confusing comment, I thought this issue is in the hpack repo.

**Comment by dbaynard** [comment]
Can this be closed now 1.9 has been released with hpack 0.31?

**Comment by borsboom** [comment]
Stack 1.9 does include the version of black with this fix.
